### PR TITLE
Asm-11268 Add gateway config resources

### PR DESCRIPTION
**Pull request by tuvia-akeyless**

### CODE REVIEW

**Review by renanaAkeyless — APPROVED** [review]

File: akeyless/resource_gateway_test.go [hunk 1]
```diff
@@ -53,3 +53,55 @@ func TestGatewayAllowedAccess(t *testing.T) {
 
 	testGatewayAllowedAccessResource(t, inputParams)
 }
+
+func TestGatewayUpdateCache(t *testing.T) {
+	t.Skip("not supported on public gateway")
```
Comment (dan-akeyless) [inline akeyless/resource_gateway_test.go:58]: If we skip the test then why do we need the test?

File: akeyless/resource_gateway_test.go [hunk 2]
```diff
@@ -53,3 +53,55 @@ func TestGatewayAllowedAccess(t *testing.T) {
...
+			enable_proactive   		= "true"
+			minimum_fetch_interval 	= "6"
+			backup_interval 		= "2"
+		}
+	`, name)
+
+	configUpdate := fmt.Sprintf(`
+		resource "akeyless_gateway_cache" "%v" {
+		}
+	`, name)
```
Comment (dan-akeyless) [inline akeyless/resource_gateway_test.go:76]: recommend using go templates

**Review by dan-akeyless — APPROVED** [review]
lgtm

File: akeyless/resource_gateway_test.go (hunk 1 above)
Comment (tuvia-akeyless) [reply akeyless/resource_gateway_test.go:58]: For a day we will set private gateway for tests, or for local testing.

**Pull request merged**